core/themes/seven/css/style.css

```diff
@@ -367,13 +367,14 @@ ul.secondary li.active a.active {
   /* Add a margin to all layouts. */
   margin: 0 20px;
 }
-.l-header {
-  /* Then add a negative margin the header the same amount. */
+body > .layout > .l-header {
```

Comment:
@quicksketch how's this work for you? Should be general enough. Specificity is pretty high, but it shouldn't ever apply where we don't want it to :)